packages/aws-cdk/lib/api/logs/logs-monitor.ts

```diff
@@ -0,0 +1,248 @@
+import * as util from 'util';
+import * as colors from 'colors/safe';
+import { print, error } from '../../logging';
+import { flatten } from '../../util/arrays';
+import { ISDK } from '../aws-auth';
+
+// how often we should read events from CloudWatchLogs
+const SLEEP = 2_000;
+
+/**
+ * Represents a CloudWatch Log Event that will be
+ * printed to the terminal
+ */
+export interface CloudWatchLogEvent {
+  /**
+   * The log event message
+   */
+  readonly message: string;
+
+  /**
+   * The name of the log group
+   */
+  readonly logGroup: string;
+
+  /**
+   * The time at which the event occurred
+   */
+  readonly timestamp: Date;
+}
+
+/**
+ * Options for adding log groups to the CloudWatchLogEventMonitor
+ */
+export interface AddLogGroupOptions {
+  /**
+   * The AWS account that the log group lives in
+   */
+  readonly account: string;
+
+  /**
+   * The AWS region that the log group lives in
+   */
+  readonly region: string;
+
+  /**
+   * The SDK for a given account
+   */
+  readonly sdk: ISDK;
+
+  /**
+   * A map of log groups and startTime in a given account
+   */
+  readonly groups: Set<string>;
+}
+
+/**
+ * Configuration tracking information on the log groups that are
+ * being monitored
+ */
+export interface LogGroupConfig {
+  /**
+   * The SDK for a given account
+   */
+  readonly sdk: ISDK;
+
+  /**
+   * A map of log groups and startTime in a given account
+   */
+  readonly groups: Map<string, number>;
```

Comment:
Actually, instead of using the notation `Map<string, number>`, can you use the notation `{ [env: string]: string }`? That will at least help by showing what the key represents.